rules/camunda-cloud/task-listener.js

```diff
@@ -17,18 +17,18 @@ module.exports = skipInNonExecutableProcess(function() {
       return;
     }
 
-    let errors = hasExtensionElement(node, 'zeebe:UserTask', node);
-
-    if (errors && errors.length) {
-      reportErrors(node, reporter, errors);
-    }
-
     const taskListeners = findExtensionElement(node, 'zeebe:TaskListeners');
 
     if (!taskListeners) {
       return;
     }
 
+    let errors = hasExtensionElement(node, 'zeebe:UserTask');
```

Comment:
This is still going to report the same error which only states that the extension element is required. It doesn't hint at the fact that this extension element is only required if another extension element is present. We have a similar error type for properties (cf. https://github.com/camunda/bpmnlint-plugin-camunda-compat/blob/main/rules/utils/error-types.js#L17). I'd consider adding a `EXTENSION_ELEMENT_DEPENDENT_REQUIRED` error type.

Reply:
I will use the error you suggested.

Reply:
We have just clarified that we will remove the check since you cannot model a job worker user task with task listeners in the UI.

Reply:
Done via https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/182/commits/df3d36366e052e0615318ef48676e436092d3c82